apps/geo/mutations.py

```diff
@@ -0,0 +1,135 @@
+
+from utils.graphene.error_types import CustomErrorType
+from utils.graphene.mutation import DeleteMutation, GrapheneMutation, generate_input_type_for_serializer
+
+RegionInputType = generate_input_type_for_serializer(
+    'RegionInputType',
+    serializer_class=RegionGqSerializer,
+)
+
+AdminLevelInputType = generate_input_type_for_serializer(
+    'AdminLevelInputType',
+    serializer_class=AdminLevelGqlSerializer,
+)
+
+
+class CreateRegion(GrapheneMutation):
+    class Arguments:
+        data = RegionInputType(required=True)
+    model = Region
+    serializer_class = RegionGqSerializer
+    result = graphene.Field(RegionDetailType)
+
+    @classmethod
+    def check_permissions(cls, info, **_):
+        return True  # global permission is always true
+    #  NOTE: Region permission is checked using serializers
+
+
+class CreateAdminLevel(GrapheneMutation):
+    class Arguments:
+        data = AdminLevelInputType(required=True)
+    model = AdminLevel
+    serializer_class = AdminLevelGqlSerializer
+    result = graphene.Field(AdminLevelType)
+
+    @classmethod
+    def check_permissions(cls, info, **_):
+        return True  # global permission is always true
+    #  NOTE: Region permission is checked using serializers
+
+
+class UpdateRegion(GrapheneMutation):
+    class Arguments:
+        data = RegionInputType(required=True)
+        id = graphene.ID(required=True)
+    model = Region
+    serializer_class = RegionGqSerializer
+    result = graphene.Field(RegionDetailType)
+
+    @classmethod
+    def check_permissions(cls, info, **_):
+        return True  # global permission is always true
+
+
+class UpdateAdminLevel(GrapheneMutation):
+    class Arguments:
+        data = AdminLevelInputType(required=True)
+        id = graphene.ID(required=True)
+    model = AdminLevel
+    serializer_class = AdminLevelGqlSerializer
+    result = graphene.Field(AdminLevelType)
+
+    @classmethod
+    def check_permissions(cls, info, **_):
+        return True  # global permission is always True
+
+
+class DeleteAdminLevel(DeleteMutation):
+    class Arguments:
+        admin_level_id = graphene.ID(required=True)
+
+    errors = graphene.List(graphene.NonNull(CustomErrorType))
+    ok = graphene.Boolean()
+    result = graphene.Field(AdminLevelType)
+
+    @staticmethod
+    def mutate(root, info, admin_level_id):
+        admin_level_qs = AdminLevel.objects.filter(
+            id=admin_level_id,
+            region__is_published=False
+        )
+        if not admin_level_qs:
+            return DeleteAdminLevel(errors=[
+                dict(
+                    field='nonFieldErrors',
+                    messages=gettext(
+                        'You should be Region owner to delete admin level or region is published'
+                    ),
+                )
+            ], ok=False)
+        admin_level_qs.delete()
+        return DeleteAdminLevel(result=admin_level_qs, errors=None, ok=True)
+
```

Comment:
Lets check owner permission, published region separately

append error in send response